src/condor_utils/file_transfer.cpp

```diff
@@ -2397,6 +2399,48 @@ FileTransfer::DoDownload( filesize_t *total_bytes_ptr, ReliSock *s)
 		SpooledJobFiles::createJobSpoolDirectory(&jobAd,desired_priv_state);
 	}
 
+	std::string outputDirectory = Iwd;
+	int directory_creation_mode = 0700;
+	bool all_transfers_succeeded = true;
+	if( final_transfer && IsServer() ) {
+		jobAd.LookupString( "OutputDirectory", outputDirectory );
+		if(! outputDirectory.empty()) {
+			std::filesystem::path outputPath( outputDirectory );
+			if(! outputPath.has_root_path()) {
+				outputDirectory = Iwd / outputPath;
+			}
+
+			if(! shadow_safe_mkdir(
+				outputDirectory, directory_creation_mode, PRIV_USER
```

Comment:
This and other calls to shadow_safe_mkdir() should use desired_priv_state instead of PRIV_USER. In practice, that will always be PRIV_USER on this code path, but it's consistent with the other prig-switching code in FileTransfer.